src/python/WMComponent/AgentStatusWatcher/ResourceControlUpdater.py

```diff
@@ -336,3 +347,34 @@ def checkTaskSlotsChanges(self, siteName, CPUBound, IOBound):
             repackSlots = int(CPUBound * self.runningRepackPercent / 100)
             pendingRepack = int(repackSlots * self.pendingSlotsTaskPercent / 100)
             self.resourceControl.insertThreshold(siteName, 'Repack', repackSlots, pendingRepack)
+
+    def updatePNNMap(self):
+        """
+        Method to make new PNN associations based on the CRIC and local
+        database PSNs/PNNs information.
+        If a PNN is changed or removed in CRIC, then it will simply be
+        logged as an error, as it cause have deeper consequences in the agent.
+        :return: None
+        """
+        cricMap = self.cricObj.PSNtoPNNMap(self.cricPattern)
+        logging.info("Retrieved %i PSN to PNNs map from CRICs", len(cricMap))
+        localMap = self.resourceControl.listSitesSlots()
+        logging.info("Retrieved %i PSN to PNNs map from local database", len(localMap))
+
+        for siteInfo in localMap:
+            siteName = siteInfo['site_name']
+            if siteName not in cricMap:
+                logging.warning("Site name: %s cannot be found in CRIC.", siteName)
+                continue
```

Comment:
I do not think we should simply continue here. We'd rather do something.  the fact that the site has disappeared  from CRIC must mean something. So I'd rather try to update the Agent's resource control and remove it from there as well. Even though it may require some more complicated code changes.

Reply:
The CRIC call uses the data-processing API, which requires an association of PNN and PSN.
Many opportunistic resources do not have a PNN, and those would fall into this warning message. Maybe I could update the message to something like "cannot be found in CRIC data-processing API"?